malariagen_data/anoph/h12_params.py

```diff
@@ -27,3 +27,38 @@
 min_cohort_size_default: base_params.min_cohort_size = 15
 
 max_cohort_size_default: base_params.max_cohort_size = 50
+
+
+class sample_query_params:
+    def __init__(
+        self,
+        sample_query: base_params.sample_query,
+        title: Optional[gplt_params.title],
+        window_size: window_size,
+        analysis: Optional[hap_params.analysis] = base_params.DEFAULT,
+        cohort_size: Optional[base_params.cohort_size] = cohort_size_default,
+        min_cohort_size: Optional[
+            base_params.min_cohort_size
+        ] = min_cohort_size_default,
+        max_cohort_size: Optional[
+            base_params.max_cohort_size
+        ] = max_cohort_size_default,
+    ) -> None:
+        self.sample_query = sample_query
+        if title:
+            self.title = title
+        else:
+            self.title = sample_query
+        self.window_size = window_size
+        self.analysis = analysis
+        self.cohort_size = cohort_size
+        self.min_cohort_size = min_cohort_size
+        self.max_cohort_size = max_cohort_size
+
+
+sample_queries: TypeAlias = Annotated[
+    Sequence[sample_query_params],
+    """
+    A set of sample queries parameters. These include actual sample queries, the title associated to each one, the window size for the analysis, the site filter analysis that needs to be used, the cohort size, the minimum and the maximum cohort sizes.
+    """,
+]
```

Comment:
I think this could be simplified, because there is only one parameter here that might need to be varied between cohorts, which is the `window_size` parameter.

I think I would suggest to remove this, and just expose these parameters (`window_size`, `analysis`, `cohort_size`, `min_cohort_size`, `max_cohort_size`) directly on the function being called by the user. For `analysis`, `cohort_size`, `min_cohort_size`, `max_cohort_size` we would expect a single value and pass this through for all cohorts. For `window_size` it could be a single value, or it could be a dictionary mapping cohort keys to window size values.

Does that make sense?

So, e.g., a user would call:

```python
ag3.plot_h12_gwss_multi_panel(
    contig="2R",
    sample_sets=[...],
    cohorts="admin1_year",
    analysis="gamb_colu_arab",
    window_size=2000,
    min_cohort_size=20,
    ...
)
```

...or if different window sizes are required, a user would call:

```python
ag3.plot_h12_gwss_multi_panel(
    contig="2R",
    sample_sets=[...],
    cohorts="admin1_year",
    analysis="gamb_colu_arab",
    window_size={"...": 2000, "...": 1000, ...},
    min_cohort_size=20,
    ...
)